datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -451,48 +456,98 @@ def _send_telemetry_counter(self, metric_name, val, scraper_config, extra_tags=N
     def _store_labels(self, metric, scraper_config):
         # If targeted metric, store labels
         if metric.name in scraper_config['label_joins']:
-            matching_label = scraper_config['label_joins'][metric.name]['label_to_match']
+            label_joins = scraper_config['label_joins'][metric.name]
+            labels_to_get = label_joins['labels_to_get']
             for sample in metric.samples:
                 # metadata-only metrics that are used for label joins are always equal to 1
                 # this is required for metrics where all combinations of a state are sent
                 # but only the active one is set to 1 (others are set to 0)
                 # example: kube_pod_status_phase in kube-state-metrics
                 if sample[self.SAMPLE_VALUE] != 1:
                     continue
-                label_dict = dict()
-                matching_value = None
-                for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
-                    if label_name == matching_label:
-                        matching_value = label_value
-                    elif label_name in scraper_config['label_joins'][metric.name]['labels_to_get']:
-                        label_dict[label_name] = label_value
-                try:
-                    if scraper_config['_label_mapping'][matching_label].get(matching_value):
-                        scraper_config['_label_mapping'][matching_label][matching_value].update(label_dict)
-                    else:
-                        scraper_config['_label_mapping'][matching_label][matching_value] = label_dict
-                except KeyError:
-                    if matching_value is not None:
-                        scraper_config['_label_mapping'][matching_label] = {matching_value: label_dict}
+
+                if 'label_to_match' in label_joins:
+                    label_dict = dict()
+                    matching_value = None
+                    matching_label = label_joins['label_to_match']
+
+                    for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
+                        if label_name == matching_label:
+                            matching_value = label_value
+                        elif label_name in labels_to_get or '*' in labels_to_get:
+                            label_dict[label_name] = label_value
+
+                    self._update_label_mapping(matching_label, matching_value, label_dict, scraper_config)
+
+                if 'labels_to_match' in label_joins:
```

Comment:
Why a second "labels_to_match" attribute? (on top of the misleading name). It probably deserves a comment to explain what we are trying to do.

Reply:
So the intention is to preserve the current behavior of matching by a single `label_to_match`. I agree it is somewhat confusing, maybe a different name?